### PR TITLE
Sanitised all SQL input and hopefully fixed all instances of builtin shadowing, also cleaned up hack command to use a loop

### PR DESCRIPTION
**Pull request by ghost**
Fixed string formatting in SQL queries to prevent injection, and replaced instances of builtin shadowing (prominently that of `id`)

also the hack command now uses a tuple of entries and a loop, saving a lot of lines of code

### CODE REVIEW

**Comment by daggy1234** [comment]
Looks great thanks a bunch!

File: dagbot/bot.py [hunk 1]
```diff
@@ -18,9 +18,9 @@
 
 
 async def get_prefix(bot, message):
-    id = message.guild.id
+    g_id = message.guild.id
```
Comment (Fogapod) [inline dagbot/bot.py:21]: You don't need local variable at all here

File: dagbot/extensions/ai.py [hunk 2]
```diff
@@ -38,9 +38,9 @@ def __init__(self, client):
         self.cleverbot.set_context(ac.DictContext(self.cleverbot))
 
     async def cog_check(self, ctx):
-        id = str(ctx.guild.id)
+        g_id = str(ctx.guild.id)
```
Comment (Fogapod) [inline dagbot/extensions/ai.py:41]: Same as above, also redundant double str cast

File: dagbot/extensions/animals.py [hunk 3]
```diff
@@ -36,9 +36,9 @@ def __init__(self, client):
         self.client = client
 
     async def cog_check(self, ctx):
-        id = str(ctx.guild.id)
+        g_id = str(ctx.guild.id)
```
Comment (Fogapod) [inline dagbot/extensions/animals.py:39]: Same as above

File: dagbot/extensions/events.py [hunk 4]
```diff
@@ -11,9 +11,9 @@ def __init__(self, bot):
     @commands.Cog.listener()
     async def on_message(self, message):
         channel = message.channel
-        id = message.guild.id
+        g_id = message.guild.id
```
Comment (Fogapod) [inline dagbot/extensions/events.py:14]: Same as above

File: dagbot/extensions/events.py [hunk 5]
```diff
@@ -38,23 +38,20 @@ async def on_message(self, message):
...
-        id = guild.id
+        g_id = guild.id
         try:
             await self.bot.pg_con.execute(
                 """
         INSERT INTO prefixesandstuff (on_message_perm,server_id,command_prefix)
         VALUES (True,$1,'do ');""",
-                str(id)
+                str(g_id)
             )
```
Comment (Fogapod) [inline dagbot/extensions/events.py:48]: Redundant str cast

File: dagbot/extensions/fun.py [hunk 6]
```diff
@@ -77,9 +77,9 @@ def __init__(self, client):
         self.client = client
 
     async def cog_check(self, ctx):
-        id = str(ctx.guild.id)
+        g_id = str(ctx.guild.id)
         for e in self.client.cogdata:
-            if str(e["serverid"]) == str(id):
+            if str(e["serverid"]) == str(g_id):
```
Comment (Fogapod) [inline dagbot/extensions/fun.py:82]: Redundant variable and str cast

File: dagbot/extensions/fun.py [hunk 7]
```diff
@@ -494,7 +494,7 @@ async def hack(self, ctx, target: discord.Member = None):
             bits = getrandbits(32)  # generates an integer with 32 random bits
             # instances an IPv4Address object from those bits
             addr = IPv4Address(bits)
```
Comment (Fogapod) [inline dagbot/extensions/fun.py:496]: Redundant addr variable

File: dagbot/extensions/fun.py [hunk 8]
```diff
@@ -503,7 +503,7 @@ async def hack(self, ctx, target: discord.Member = None):
             # .compressed contains the short version of the IPv6 address
             # str(addr) always returns the short address
```
Comment (Fogapod) [inline dagbot/extensions/fun.py:504]: Redundant addr variable

File: dagbot/extensions/games.py [hunk 9]
```diff
@@ -611,9 +611,9 @@ def __init__(self, bot):
             self.not_onion_headlines = f.read().splitlines()
 
     async def cog_check(self, ctx):
-        id = str(ctx.guild.id)
+        g_id = str(ctx.guild.id)
         for e in self.bot.cogdata:
-            if str(e["serverid"]) == str(id):
+            if str(e["serverid"]) == str(g_id):
```
Comment (Fogapod) [inline dagbot/extensions/games.py:616]: Redundant variable and double str cast

File: dagbot/extensions/games.py [hunk 10]
```diff
@@ -709,7 +709,7 @@ async def mcq(self):
         level = file["results"][0]["difficulty"]
         q = file["results"][0]["question"]
         ca = file["results"][0]["correct_answer"]
-        list = file["results"][0]["incorrect_answers"]
+        list_ = file["results"][0]["incorrect_answers"]
```
Comment (Fogapod) [inline dagbot/extensions/games.py:712]: Confusing name

File: dagbot/extensions/whysomart.py [hunk 11]
```diff
@@ -436,12 +436,12 @@ async def fact(self, message):
     @commands.command(cooldown_after_parsing=True)
     async def define(self, ctx, *, word: str):
         await ctx.trigger_typing()
-        dict = dictionary.meaning(word)
-        k = dict.keys()
+        dict_ = dictionary.meaning(word)
```
Comment (Fogapod) [inline dagbot/extensions/whysomart.py:439]: Confusing variable name

File: dagbot/extensions/whysomart.py [hunk 12]
```diff
@@ -45,9 +45,9 @@ def __init__(self, client):
         self.client = client
 
     async def cog_check(self, ctx):
-        id = str(ctx.guild.id)
+        g_id = str(ctx.guild.id)
         for e in self.client.cogdata:
-            if str(e["serverid"]) == str(id):
+            if str(e["serverid"]) == str(g_id):
```
Comment (Fogapod) [inline dagbot/extensions/whysomart.py:50]: Code duplication, redundant variable and str cast

File: dagbot/extensions/util.py [hunk 13]
```diff
@@ -165,9 +165,9 @@ def __init__(self, client):
         self.googlethingy = async_cse.Search(self.client.data["gapikey"])
 
     async def cog_check(self, ctx):
-        id = str(ctx.guild.id)
+        g_id = str(ctx.guild.id)
         for e in self.client.cogdata:
-            if str(e["serverid"]) == str(id):
+            if str(e["serverid"]) == str(g_id):
```
Comment (Fogapod) [inline dagbot/extensions/util.py:170]: Code duplication, redundant variable and str cast

File: dagbot/extensions/text.py [hunk 14]
```diff
@@ -26,9 +26,9 @@ def __init__(self, client):
         self.cipher_suite = Fernet(Fernet.generate_key())
 
     async def cog_check(self, ctx):
-        id = str(ctx.guild.id)
+        g_id = str(ctx.guild.id)
         for e in self.client.cogdata:
-            if str(e["serverid"]) == str(id):
+            if str(e["serverid"]) == str(g_id):
```
Comment (Fogapod) [inline dagbot/extensions/text.py:31]: Code duplication, redundant variable and str cast

File: dagbot/extensions/tags.py [hunk 15]
```diff
@@ -159,7 +159,7 @@ async def delete(self, ctx, *, name):
 
             if len(y) == 0:
                 return await ctx.send("Tag does not exist")
-            elif (str(y[0]["author"])) == id:
+            elif (str(y[0]["author"])) == auth_id:
                 await self.bot.pg_con.execute(
```
Comment (Fogapod) [inline dagbot/extensions/tags.py:163]: auth?

File: dagbot/extensions/tags.py [hunk 16]
```diff
@@ -9,9 +9,9 @@ def __init__(self, bot):
         self.bot = bot
 
     async def cog_check(self, ctx):
-        id = str(ctx.guild.id)
+        g_id = str(ctx.guild.id)
         for e in self.bot.cogdata:
-            if str(e["serverid"]) == str(id):
+            if str(e["serverid"]) == str(g_id):
```
Comment (Fogapod) [inline dagbot/extensions/tags.py:14]: Code duplication, redundant variable and str cast

**Review by Fogapod — COMMENTED** [review]
That's a lot of code duplication

**Pull request merged**